src/main/java/org/jvnet/hudson/test/recipes/WithPlugin.java

```diff
@@ -89,6 +90,9 @@ public void setup(JenkinsRule jenkinsRule, WithPlugin recipe) throws Exception {
         public void decorateHome(JenkinsRule jenkinsRule, File home) throws Exception {
             for (String plugin : a.value()) {
                 URL res = getClass().getClassLoader().getResource("plugins/" + plugin);
+                if (res == null) {
+                    throw new IOException("Cannot find resource for plugin " + plugin);
```

Comment:
This only applies to tests actually annotated with `@WithPlugin`. No effect on tests assuming plugin existence due to a `test`-scoped dependency.

Reply:
Yes. It is not really related to the defect. Just better diagnostics for another issue I hit. Can detach it to a separate PR if you prefer

Reply:
I think I've been impacted by this issue too doing some tests in support-core plugin. Here the folder _plugins_ doesn't exist: https://github.com/jenkinsci/support-core-plugin/blob/54fde93fa28ac1c69adc0a2a6c4962ca2c629b3c/src/main/java/com/cloudbees/jenkins/support/impl/AboutJenkins.java#L837

Reply:
IIRC the plugin registry is just not initialized in the Test plugin manager